snippets/dig.md

```diff
@@ -0,0 +1,56 @@
+### dig
+
+Return the value in a nested JSON object by the key.
+
+Given the key name (or target), it will look up the key in the object recursively and return the first value if found.
```

Comment:
Please explain the methods used here, such as recursion of `dig` and what happens in it.